### PR TITLE
Fix fileserver backends __opts__ overwritten by _pillar

### PR DESCRIPTION
**Pull request by bersace**
Refs #22941 
Supersedes #22942 

This time, I only override `__opts__` globals so that `file_roots` is not `pillar_roots`.

What do you think of it ? :)

### CODE REVIEW

**Comment by bersace** [comment]
Jenkins OK! Congrats !

File: salt/master.py [hunk 1]
```diff
@@ -768,7 +768,7 @@ def __setup_fileserver(self):
         '''
         Set the local file objects from the file server interface
         '''
-        fs_ = salt.fileserver.Fileserver(self.opts)
+        self.fs_ = fs_ = salt.fileserver.Fileserver(self.opts)
```
Comment (thatch45) [inline salt/master.py]: Can this just be:
self.fs_ = salt.fileserver.Fileserver(self.opts)
and then change the fs refs below?

**Comment by thatch45** [comment]
This should be a lot better!

**Comment by thatch45** [comment]
@cachedout can you take a look here? We need to figure out a cleaner way to handle the pillar data overrideas

File: salt/master.py (hunk 1 above)
Comment (bersace) [inline salt/master.py]: Done :)

**Comment by bersace** [comment]
Lol. The timeout test did not timeout, but the other did timeout :)

**Comment by bersace** [comment]
Any news on CI on it ? Do i have to retrigger the build ?

**Comment by rallytime** [comment]
@bersace Sorry for not responding sooner here. From a testing perspective, both of those test failures are known to be flaky tests, so those failures are not related to this code change. I've added the "Tests Passed" label.

**Comment by jfindlay** [comment]
@thatch45, does this pull request look ready to go?

**Comment by thatch45** [comment]
Sorry about the wait, I have been behind on merged this week.
@cachedout please be aware of this one

**Pull request merged**